packages/appsync-modelgen-plugin/src/utils/process-connections-v2.ts

```diff
@@ -124,18 +128,19 @@ export function processConnectionsV2(
   field: CodeGenField,
   model: CodeGenModel,
   modelMap: CodeGenModelMap,
-  shouldUseModelNameFieldInHasManyAndBelongsTo: boolean
+  shouldUseModelNameFieldInHasManyAndBelongsTo: boolean = false,
+  isCustomPKEnabled: boolean = false
 ): CodeGenFieldConnection | undefined {
   const connectionDirective = field.directives.find(d => d.name === 'hasOne' || d.name === 'hasMany' || d.name === 'belongsTo');
 
   if (connectionDirective) {
     switch (connectionDirective.name) {
       case 'hasOne':
-        return processHasOneConnection(field, model, modelMap, connectionDirective);
+        return processHasOneConnection(field, model, modelMap, connectionDirective, isCustomPKEnabled);
```

Comment:
I would assume this list of arguments is going to grow with time, we need a more scalable approach to pass in the customizations to these `process<directive>` methods. Something to think about.